cmd/uplink/main.go

```diff
@@ -91,18 +93,34 @@ func cmdSetup(cmd *cobra.Command, args []string) (err error) {
 		return err
 	}
 
+	awsAccessCreds, awsSecretCreds, err := getGeneralAwsCreds()
+	if err != nil {
+		return err
+	}
+
 	o := map[string]interface{}{
 		"cert-path":       setupCfg.Identity.CertPath,
 		"key-path":        setupCfg.Identity.KeyPath,
 		"api-key":         setupCfg.APIKey,
 		"pointer-db-addr": setupCfg.SatelliteAddr,
 		"overlay-addr":    setupCfg.SatelliteAddr,
+		"access-key":      awsAccessCreds,
+		"secret-key":      awsSecretCreds,
 	}
 
 	return process.SaveConfig(runCmd.Flags(),
 		filepath.Join(setupCfg.BasePath, "config.yaml"), o)
 }
 
+func getGeneralAwsCreds() (string, string, error) {
+	var buf [20]byte
+	_, err := rand.Read(buf[:])
+	if err != nil {
+		return "", "", err
+	}
+	return base58.Encode(buf[:]), base58.Encode(buf[1:]), nil
```

Comment:
It looks like the secret key is the access key without the first byte. This is very weak from security point of view. If I know the access key, I automatically know the secret key too. And the access key is part of the URL generated by the aws cli preset command (or when sharing an object via the minio browser).

It would be best to avoid any relationship between the access key and the secret key. Like generate a separate random []byte for each of them.

Reply:
will do